algo/src/main/java/org/neo4j/graphalgo/impl/pagerank/ComputeStepFactory.java

```diff
@@ -3,11 +3,15 @@
 import org.neo4j.graphalgo.api.*;
 import org.neo4j.graphalgo.core.utils.paged.AllocationTracker;
 
+import java.util.concurrent.ExecutorService;
+
 
 public interface ComputeStepFactory {
     ComputeStep createComputeStep(double dampingFactor, int[] sourceNodeIds, RelationshipIterator relationshipIterator, Degrees degrees, RelationshipWeights relationshipWeights, int partitionCount, int start, double[] aggregatedDegrees);
 
     HugeComputeStep createHugeComputeStep(double dampingFactor, long[] sourceNodeIds, HugeRelationshipIterator relationshipIterator, HugeDegrees degrees, HugeRelationshipWeights relationshipWeights, AllocationTracker tracker, int partitionCount, long start, double[] aggregatedDegrees);
+
+    double[] degrees(Graph graph, ExecutorService executor, int concurrency);
```

Comment:
feels like in the wrong place?

Reply:
why is degrees double by default?

Reply:
I've renamed that class now. The reason it's there is because we don't want to calculate it for the non weighted version.

The reason it's double is because it's being used for weighted degree and users can set the weight to be a double value, not just integer..